app/views/materials/compare.html.erb

```diff
@@ -0,0 +1,44 @@
+<div class="jumbotron">
+  <form>
+    <% @materials.each_with_index do |material, index| %>
+      <div class="row">
+        <div class="col mb-4">
+          <%= label_tag "material #{index+1}" %>
+          <%= text_field nil, "material_ids[]", value: params[:material_ids][index], class: "form-control", placeholder: "UUID" %>
+        </div>
+      </div>
+    <% end %>
+    <div class="row">
+      <div class="col">
+        <%= submit_tag "Compare", class: "btn btn-primary", name: "compare" %>
+        <%= submit_tag "Add material", class: "btn btn-primary", name: "add" %>
+      </div>
+    </div>
+  </form>
+</div>
+
+<% @materials = @materials.compact %>
+
+<% if @materials.size  > 0 %>
+  <table class="table table-striped">
+    <thead>
+      <tr>
+        <th>Property</th>
+        <% @materials.each do |material| %>
+          <th><%= material.name %></th>
+        <% end %>
+      </tr>
+    </thead>
+    <tbody>
+      <% attributes_by_material_function(@materials[0]).each do |attr, val| %>
+      <% values = @materials.collect{ |material| material.attributes.fetch(attr)} %>
```

Comment:
Would it make sense to move this into the material comparator? Could convert the list of materials into a hash of attributes mapped to the list of values.

Reply:
I'm not too eager to put the list of attributes in the material_comparator, it would mean changing the show method as well or duplicating the list. I've moved the logic into the material_helper though, so let me know what you think now?